src/expression/node/FunctionNode.js

```diff
@@ -77,22 +76,22 @@ export const createFunctionNode = /* #__PURE__ */ factory(name, dependencies, ({
     }
 
     // compile arguments
-    const evalArgs = map(this.args, function (arg) {
-      return arg._compile(math, argNames)
-    })
+    const evalArgs = this.args.map((arg) => arg._compile(math, argNames))
```

Comment:
I re-wrote a lot of this method just to understand it all. 

Nested trinaries, and keeping track of `this`, `fn.apply`, BYO `map` function. I hope you don't mind.